### PR TITLE
rampis: rename interfaces on edgerouter-X

### LINKED ISSUE #15643: ramps: eth1 fails to initialise on Edgerouter X
### Describe the bug

I just updated my er-x build to latest main and eth1 has stopped working. This would seem to be a regression in the past few weeks and I do notice some patches around ethernet support on rampis.

### OpenWrt version

r26463+81-a1a9572f43

### OpenWrt release

SNAPSHOT

### OpenWrt target/subtarget

ramips/mt7621

### Device

Ubiquiti EdgeRouter X

### Image kind

Self-built image

### Steps to reproduce

1. sysupgrade to snapshot at commit c67d5189a405
2. reboot

### Actual behaviour

eth1 is missing, kernel logs show:
```
Thu Jun  6 20:40:48 2024 kern.info kernel: [    3.233498] mt7530-mdio mdio-bus:1f: MT7530 adapts as multi-chip module
Thu Jun  6 20:40:48 2024 kern.info kernel: [    3.273510] mt7530-mdio mdio-bus:1f: configuring for fixed/rgmii link mode
Thu Jun  6 20:40:48 2024 kern.info kernel: [    3.288355] mt7530-mdio mdio-bus:1f: Link is Up - 1Gbps/Full - flow control rx/tx
Thu Jun  6 20:40:48 2024 kern.info kernel: [    3.293786] mt7530-mdio mdio-bus:1f eth1 (uninitialized): PHY [mt7530-0:01] driver [MediaTek MT7530 PHY] (irq=21)
Thu Jun  6 20:40:48 2024 kern.err kernel: [    3.324660] mtk_soc_eth 1e100000.ethernet eth0: error -17 registering interface eth1
```

### Expected behaviour

eth1 should exist and be part of lan

### Additional info

My custom build only includes this PR https://github.com/openwrt/openwrt/pull/15194, to allow 6.6 kernel no other customisation.

### Diffconfig

```text
./scripts/diffconfig.sh

CONFIG_TARGET_ramips=y
CONFIG_TARGET_ramips_mt7621=y
CONFIG_TARGET_ramips_mt7621_DEVICE_ubnt_edgerouter-x=y
CONFIG_GETDNS_ENABLE_STUB_ONLY=y
CONFIG_LIBCURL_COOKIES=y
CONFIG_LIBCURL_FILE=y
CONFIG_LIBCURL_FTP=y
CONFIG_LIBCURL_HTTP=y
CONFIG_LIBCURL_MBEDTLS=y
CONFIG_LIBCURL_NGHTTP2=y
CONFIG_LIBCURL_NO_SMB="!"
CONFIG_LIBCURL_PROXY=y
CONFIG_LIBCURL_UNIX_SOCKETS=y
CONFIG_OPENSSL_ENGINE=y
CONFIG_OPENSSL_PREFER_CHACHA_OVER_GCM=y
CONFIG_OPENSSL_WITH_ASM=y
CONFIG_OPENSSL_WITH_CHACHA_POLY1305=y
CONFIG_OPENSSL_WITH_CMS=y
CONFIG_OPENSSL_WITH_DEPRECATED=y
CONFIG_OPENSSL_WITH_ERROR_MESSAGES=y
CONFIG_OPENSSL_WITH_IDEA=y
CONFIG_OPENSSL_WITH_MDC2=y
CONFIG_OPENSSL_WITH_PSK=y
CONFIG_OPENSSL_WITH_SEED=y
CONFIG_OPENSSL_WITH_SRP=y
CONFIG_OPENSSL_WITH_TLS13=y
CONFIG_OPENSSL_WITH_WHIRLPOOL=y
CONFIG_PACKAGE_cgi-io=y
CONFIG_PACKAGE_curl=m
CONFIG_PACKAGE_ddns-scripts=m
CONFIG_PACKAGE_ddns-scripts-cloudflare=m
CONFIG_PACKAGE_ddns-scripts-services=m
CONFIG_PACKAGE_getdns=m
CONFIG_PACKAGE_kmod-crypto-chacha20poly1305=m
CONFIG_PACKAGE_kmod-crypto-lib-chacha20=m
CONFIG_PACKAGE_kmod-crypto-lib-chacha20poly1305=m
CONFIG_PACKAGE_kmod-crypto-lib-curve25519=m
CONFIG_PACKAGE_kmod-crypto-lib-poly1305=m
CONFIG_PACKAGE_kmod-udptunnel4=m
CONFIG_PACKAGE_kmod-udptunnel6=m
CONFIG_PACKAGE_kmod-wireguard=m
CONFIG_PACKAGE_libatomic=m
CONFIG_PACKAGE_libcurl=m
CONFIG_PACKAGE_liblua=m
CONFIG_PACKAGE_liblucihttp=y
CONFIG_PACKAGE_liblucihttp-lua=m
CONFIG_PACKAGE_liblucihttp-ucode=y
CONFIG_PACKAGE_libnghttp2=m
CONFIG_PACKAGE_libopenssl=m
CONFIG_PACKAGE_libubus-lua=m
CONFIG_PACKAGE_libyaml=m
CONFIG_PACKAGE_lua=m
CONFIG_PACKAGE_luci=y
CONFIG_PACKAGE_luci-app-ddns=m
CONFIG_PACKAGE_luci-app-firewall=y
CONFIG_PACKAGE_luci-app-opkg=y
CONFIG_PACKAGE_luci-base=y
CONFIG_PACKAGE_luci-lib-base=m
CONFIG_PACKAGE_luci-lib-ip=m
CONFIG_PACKAGE_luci-lib-jsonc=m
CONFIG_PACKAGE_luci-lib-nixio=m
CONFIG_PACKAGE_luci-light=y
CONFIG_PACKAGE_luci-lua-runtime=m
CONFIG_PACKAGE_luci-mod-admin-full=y
CONFIG_PACKAGE_luci-mod-network=y
CONFIG_PACKAGE_luci-mod-status=y
CONFIG_PACKAGE_luci-mod-system=y
CONFIG_PACKAGE_luci-proto-ipv6=y
CONFIG_PACKAGE_luci-proto-ppp=y
CONFIG_PACKAGE_luci-proto-wireguard=m
CONFIG_PACKAGE_luci-theme-bootstrap=y
CONFIG_PACKAGE_rpcd=y
CONFIG_PACKAGE_rpcd-mod-file=y
CONFIG_PACKAGE_rpcd-mod-iwinfo=y
CONFIG_PACKAGE_rpcd-mod-luci=y
CONFIG_PACKAGE_rpcd-mod-rrdns=y
CONFIG_PACKAGE_rpcd-mod-ucode=y
CONFIG_PACKAGE_stubby=m
CONFIG_PACKAGE_ucode-mod-html=y
CONFIG_PACKAGE_ucode-mod-lua=m
CONFIG_PACKAGE_ucode-mod-math=y
CONFIG_PACKAGE_uhttpd=y
CONFIG_PACKAGE_uhttpd-mod-ubus=y
CONFIG_PACKAGE_wireguard-tools=m
# CONFIG_PACKAGE_kmod-crypto-kpp is not set
```

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

### PR DESCRIPTION
**Pull request by darkxst**
With linux-6.6 there is a naming conflict with the eth1 port, when dsa brings up the switch it was already allocated to gmac. As a result eth1 port ends up missing and can't be used.

Rename the ports to lan1-4 and wan, which avoids the name conflict and is more consistent with other ramips targets. Network rules will need to be updated.

I haven't added a compat bump here, as my other PR modifying the Edgerouter-X partition layout bumped to compat 2.0. This was only merged into `main` today.

Fixes: #15643

logs showing error:
```
Thu Jun  6 20:40:48 2024 kern.info kernel: [    3.233498] mt7530-mdio mdio-bus:1f: MT7530 adapts as multi-chip module
Thu Jun  6 20:40:48 2024 kern.info kernel: [    3.273510] mt7530-mdio mdio-bus:1f: configuring for fixed/rgmii link mode
Thu Jun  6 20:40:48 2024 kern.info kernel: [    3.288355] mt7530-mdio mdio-bus:1f: Link is Up - 1Gbps/Full - flow control rx/tx
Thu Jun  6 20:40:48 2024 kern.info kernel: [    3.293786] mt7530-mdio mdio-bus:1f eth1 (uninitialized): PHY [mt7530-0:01] driver [MediaTek MT7530 PHY] (irq=21)
Thu Jun  6 20:40:48 2024 kern.err kernel: [    3.324660] mtk_soc_eth 1e100000.ethernet eth0: error -17 registering interface eth1
```

### CODE REVIEW

File: target/linux/ramips/mt7621/base-files/etc/board.d/02_network [hunk 1]
```diff
@@ -157,11 +157,8 @@ ramips_setup_interfaces()
 	tplink,mr600-v2-eu)
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
 		;;
-	ubnt,edgerouter-x)
-		ucidef_set_interfaces_lan_wan "eth1 eth2 eth3 eth4" "eth0"
-		;;
```
Comment (namiltd) [inline target/linux/ramips/mt7621/base-files/etc/board.d/02_network:162]: Why did you remove lines 160 to 162?
Shouldn't it be: ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4" "wan" ?

**Comment by robimarko** [comment]
Better rename the DSA port as Edgerouter has printed EthN names on the case

**Comment by darkxst** [comment]
> Better rename the DSA port as Edgerouter has printed EthN names on the case

Yes it has printed ethN naming printed on the case, however I personally would prefer a working port over matching naming.

It is not that simple to keep the existing ethN naming, there have been proposals to keep the ethN naming but require non-standard dts fields and custom scripting to work, all for just this device.

https://github.com/openwrt/openwrt/issues/15643#issuecomment-2306689200
 
Edgerouter-X devices seem to be the only rampis device with ethN labelling. There is one other rampis target (deco AP) with ethN naming and probably impacted the same issue, but it doesnt have any labelling on the case.

If it is really preferred I can change to using the `openwrt,netdev-name` labels and renaming script. However I do think its simpler and cleaner to just ignore the labelling on the case?

File: target/linux/ramips/dts/mt7621_ubnt_edgerouter-x.dtsi [hunk 2]
```diff
@@ -29,33 +29,33 @@
...
 		port@0 {
 			status = "okay";
-			label = "eth0";
+			label = "wan";
 		};
 
 		port@1 {
 			status = "okay";
-			label = "eth1";
+			label = "lan1";
```
Comment (DragonBluep) [inline target/linux/ramips/dts/mt7621_ubnt_edgerouter-x.dtsi:37]: Why not use uppercase, EthN or ETHN. I guess the label is handled by C language.

Comment (darkxst) [reply target/linux/ramips/dts/mt7621_ubnt_edgerouter-x.dtsi:37]: This would work, but looks a bit odd especially when wan ends up eth0 (but Eth0 in web ui) and lan ports as EthN. could also be a bit error prone if people try to make rules for eth1 instead of Eth1?

```
3: eth0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN qlen 1000
    link/ether ac:8b:a9:a4:56:a0 brd ff:ff:ff:ff:ff:ff
4: Eth1@dsa: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br-lan state LOWERLAYERDOWN qlen 1000
    link/ether ac:8b:a9:a4:56:a1 brd ff:ff:ff:ff:ff:ff
5: Eth2@dsa: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP qlen 1000
    link/ether ac:8b:a9:a4:56:a2 brd ff:ff:ff:ff:ff:ff
6: Eth3@dsa: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br-lan state LOWERLAYERDOWN qlen 1000
    link/ether ac:8b:a9:a4:56:a3 brd ff:ff:ff:ff:ff:ff
7: Eth4@dsa: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue master br-lan state LOWERLAYERDOWN qlen 1000
    link/ether ac:8b:a9:a4:56:a4 brd ff:ff:ff:ff:ff:ff
 ```

**Comment by robimarko** [comment]
@darkxst Doesnt simply adding a label to the GMAC as well fix this?

**Comment by darkxst** [comment]
> @darkxst Doesnt simply adding a label to the GMAC as well fix this?

Perhaps, but I dont quite get how GMAC relabelling works or atleast where the `eth1` label comes from?

Already:
gmac0 has label "dsa"
gmac1 has label "eth0" and ends up "eth0" as wan port


So somehow gmac0 become eth1 before the dsa ports are relabelled and then resulting in name conflict on eth1?
